src/code/z_bgcheck.c

```diff
@@ -1,432 +1,4713 @@
+        for (j = i + 1; j < 3; j++) {
+            if (vtxIdA[i] == vtxIdB[j]) {
+                if (count == 0) {
+                    *outVtxId0 = vtxIdA[i];
+                } else if (count == 1) {
+                    *outVtxId1 = vtxIdA[i];
+                }
+                count++;
+            }
+        }
+    }
+    return count;
+}
+
+s16 CollisionPoly_GetMinY(CollisionPoly* poly, Vec3s* vertices) {
+    s16 phi_a3;
+    s32 a;
+    s32 b;
+    s32 c;
+
+    a = COLPOLY_VTX_INDEX(poly->flags_vIA);
+    b = COLPOLY_VTX_INDEX(poly->flags_vIB);
+    c = poly->vIC;
+
+    phi_a3 = vertices[a].y;
+    if (vertices[b].y < phi_a3) {
+        phi_a3 = vertices[b].y;
+    }
+    if (phi_a3 < vertices[c].y) {
+        return phi_a3;
+    }
+    return vertices[c].y;
+}
+
+void CollisionPoly_GetNormalF(CollisionPoly* poly, f32* nx, f32* ny, f32* nz) {
+    *nx = COLPOLY_GET_NORMAL(poly->normal.x);
+    *ny = COLPOLY_GET_NORMAL(poly->normal.y);
+    *nz = COLPOLY_GET_NORMAL(poly->normal.z);
+}
+
+/**
+ * Compute transform matrix mapping +y (up) to the collision poly's normal
+ */
+void func_800C0094(CollisionPoly* poly, f32 tx, f32 ty, f32 tz, MtxF* dest) {
+    f32 nx;
+    f32 ny;
+    f32 nz;
+    s32 pad;
+    f32 z_f14;
+    f32 phi_f14;
+    f32 phi_f12;
+    f32 inv_z_f14;
+
+    if (poly == NULL) {
+        return;
+    }
+    CollisionPoly_GetNormalF(poly, &nx, &ny, &nz);
+
+    z_f14 = sqrtf(SQ(ny) + SQ(nz));
+    if (!IS_ZERO(z_f14)) {
+        inv_z_f14 = 1.0f / z_f14;
+        phi_f14 = ny * inv_z_f14;
+        phi_f12 = nz * inv_z_f14;
+    } else {
+        phi_f14 = 1.0f;
+        phi_f12 = 0.0f;
+    }
+    dest->xx = z_f14;
+    dest->xy = (-nx) * phi_f14;
+    dest->xz = (-nx) * phi_f12;
+    dest->yx = nx;
+    dest->yy = ny;
+    dest->yz = nz;
+    dest->zx = 0.0f;
+    dest->zy = -phi_f12;
+    dest->zz = phi_f14;
+    dest->wx = tx;
+    dest->wy = ty;
+    dest->wz = tz;
+    dest->xw = 0.0f;
+    dest->yw = 0.0f;
+    dest->zw = 0.0f;
+    dest->ww = 1.0f;
+}
+
+f32 CollisionPoly_GetPointDistanceFromPlane(CollisionPoly* poly, Vec3f* point) {
+    return (poly->normal.x * point->x + poly->normal.y * point->y + poly->normal.z * point->z) * COLPOLY_NORMAL_FRAC +
+           poly->dist;
+}
+
+void CollisionPoly_GetVertices(CollisionPoly* poly, Vec3s* vtxList, Vec3f* dest) {
+    BgCheck_Vec3sToVec3f(&vtxList[COLPOLY_VTX_INDEX(poly->flags_vIA)], &dest[0]);
+    BgCheck_Vec3sToVec3f(&vtxList[COLPOLY_VTX_INDEX(poly->flags_vIB)], &dest[1]);
+    BgCheck_Vec3sToVec3f(&vtxList[poly->vIC], &dest[2]);
+}
+
+void CollisionPoly_GetVerticesByBgId(CollisionPoly* poly, s32 bgId, CollisionContext* colCtx, Vec3f* dest) {
+    Vec3s* vtxList;
+
+    if (poly == NULL || bgId > BG_ACTOR_MAX || dest == NULL) {
+        if (dest != NULL) {
+            // @bug: dest[2] x and y are not set to 0
+            dest[0].x = dest[0].y = dest[0].z = dest[1].x = dest[1].y = dest[1].z = dest[2].z = 0.0f;
+            return;
+        }
```

Comment:
```suggestion
            dest[0].x = dest[0].y = dest[0].z = dest[1].x = dest[1].y = dest[1].z = dest[2].z = 0.0f;
        }
```
Is this return needed? Looks like the remainder of the function should be equivalent